docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -7,266 +7,200 @@
-    A few observations on this:
-    - we do not strictly have to call the enclave via ECC. ECCManager could also directly call
-      enclave.so. this would make this creation and key-management potentially easier?
-    - the quote generation is not driven from within the enclave as currently shown above but from
-      the outside!
-    I think from a high-level there should be following three high-level calls either in
-    ECC or ECCMAnager depending on above:
-    - EnclaveCreate (key-gen & imprint of context, i.e., current create/binding flows
-    - EnclaveProof: gets org endorsement-sig, creates report, then quote
-      and, if epid, gets IAS report. 
-    - EnclavePersist: returns sealed state (important for security that this is called only after
-      EnclaveProof [which we might want to enforce using some state-machine?] )
-  end note
-  return ATTESTATION
+  return Credentials, Sealed_Credentials
 
-  |||
+  E_Manager -> E_Manager : store Sealed_Credentials
 
-  Manager1 -> Manager1 : mark ECC_ID as installed
-  note right
-    Manager1 maintains a list of
-    locally "installed/created" enclaves
-  end note
-  return ATTESTATION, ECC_S_PK / error
+  return result /' Admin -> E_Manager '/
 
   deactivate Admin
-  |||
+
+'  note over Admin1 #FFaa00
+'    **MST**: From an attestation generalization perspective i think there should be just a single
+'    call ""GetAttestationProof"" or alike rather than a separate place where you get quote
+'    and another where you get the registration report which requires some re-organization of calls.
+'    I think from a high-level there should be following three high-level calls either in
+'    ECC or ECCMAnager depending on above:
+'    - EnclaveCreate (key-gen & imprint of context, i.e., current create/binding flows
+'    - EnclaveProof: gets org endorsement-sig, creates report, then quote
+'      and, if epid, gets IAS report. 
+'    - EnclavePersist: returns sealed state (important for security that this is called only after
+'      EnclaveProof [which we might want to enforce using some state-machine?] )
+'  end note
+'  return ATTESTATION
 
 end
 
-group register attestation
+group Org-Enclave binding/certification (to be implemented)
+  Admin -> E_Manager ++: certify(CC_Id, Cert_Parameters)
+  E_Manager -> ECC1 ++: getCSR(Cert_Parameters)
+  ECC1 -> Enclave1 ++: getCSR(Cert_Parameters)
+  return csr /' ECC1 -> Enclave1 '/
+  return csr /' E_Manager -> ECC1 '/
+  E_Manager -> E_Manager : generate enclave certificate (certified by Org_1)
+  return certificate
+  note over Admin, E_Manager #FFaa00
+    -Org_1 takes "ownership" of this enclave,
+    -the certificate should be made available to the chaincode endorsement plugin,
+     so to include it in the responses (for eventual validation) 
+  end note
+end group
 
+group enclave registration
   activate Admin
+  Admin -> E_Manager ++: register
+  E_Manager -> ERCC1 ++: register Credentials
```

Comment:
maybe, but we should deal with this later because:
1. we don't have the certificate
2. I am not sure about that. If the certificate is usually appended to responses (or anywhere else clients/peers can find it), do we also need it here? -- unless we replace the public key.

Reply:
as mentioned above, either we do certificates and then do the whole thing or we drop them completely. Putting them only half-way doesn't seem useful but rather confusing?

Reply:
b9be319f68fe8af7ccf582098fd1ea06602809de